### PR TITLE
Add Ovirt platform type

### PR DESCRIPTION
**Pull request by rgolangh**
This will be the start of adding oVirt as a platform provider for openshift.
All work in progress is focusing IPI, and is relying on metal-3 work to self contain dns and load balancing. WIP can be found here - https://github.com/rgolangh/installer

Along with this we also need:

- Machine config operator to include ovirt - https://github.com/openshift/machine-config-operator/issues/766
- oVirt to support ignition - patches merged to master - https://gerrit.ovirt.org/#/c/99372/
- oVirt to implement cluster-api-provider - https://github.com/ovirt/cluster-api-provider-ovirt

### CODE REVIEW

**Comment by cgwalters** [comment]
Worth adding some high level details here of the platform.

Are there any plans to implement [machine API](https://github.com/openshift/machine-api-operator) for oVirt?  Or will this feel more like a baremetal-in-virt platform, i.e. node provisioning is admin controlled?

Do you have any work on an RHCOS image?  How does Ignition get userdata?

File: config/v1/types_infrastructure.go [hunk 1]
```diff
@@ -91,6 +91,9 @@ const (
 
 	// VSpherePlatformType represents VMWare vSphere infrastructure.
 	VSpherePlatformType PlatformType = "VSphere"
+
+	// OvirtPlatformType represent oVirt/RHV infrastructure.
+	OvirtPlatformType PlatformType = "Ovirt"
```
Comment (wking) [inline config/v1/types_infrastructure.go]: `OVirtPlatformType PlatformType = "oVirt"`?

Comment (wking) [reply config/v1/types_infrastructure.go]: > I hope that when we compare platform we ignore cases.

Most code I've uses `switch` or equality checks without `Lower` wrapping.  If we expect folks to use case-insensitive comparison (I'm fine either way), we'll probably want to land strong language about it here.  And also find a way to test non-canonical casing vs. our e2e suite.

**Comment by rgolangh** [comment]
ping?

**Comment by rgolangh** [comment]
ping

File: config/v1/types_infrastructure.go [hunk 2]
```diff
@@ -91,6 +91,9 @@ const (
 
 	// VSpherePlatformType represents VMWare vSphere infrastructure.
 	VSpherePlatformType PlatformType = "VSphere"
+
+	// OvirtPlatformType represent oVirt/RHV infrastructure.
+	OvirtPlatformType PlatformType = "oVirt"
```
Comment (deads2k) [inline config/v1/types_infrastructure.go]: you need to add this to the doc on the field below.

**Comment by deads2k** [comment]
reapplying from @abhinavdahiya 

/lgtm
/approve

**Comment by deads2k** [comment]
You need to regenerate

**Pull request merged**